Fix(eos_cli_config_gen)!: Avoid generating invalid configuration for traffic policies (#4266)
aristanetworks/ansible-avd: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md

With AVD 5.0.0, the default value for `type` in `ethernet_interfaces` and `port_channel_interfaces` is no longer supported. The `type` key must now be explicitly defined in the input variables if it is needed in the configuration and documentation.

### Updated the eos J2 template for traffic-policies

The change has been incorporated into `traffic-policies.j2` eos template.

> In AVD 4.0.0 protocol UDP under traffic policies was accepting the flags and if the source/destination port was defined for TCP/UDP protocol then unable to define the source/destination field-set.

Reviewer: I think changes in template is not needed to document in porting-guide.

+       protocol: unmatched
```

### Fix issues with flags, source/destination ports, and field set for TCP/UDP protocol under traffic-policies

The change has been incorporated into `traffic-policies.j2` eos template.

> In AVD 4.0.0 protocol UDP under traffic policies was accepting the flags and if the source/destination port was defined for TCP/UDP protocol then unable to define the source/destination field-set.

Reviewer 2: ```suggestion
In AVD 4.0.0, configuring the `flags` for `UDP` protocol under traffic-policies was allowed and the configuration for `source/destination port` and `source/destination field-set` together was not generated for `TCP/UDP` protocol.
```

> As of AVD 5.0.0 protocol UDP under traffic policies is not accepting the flags and now both the source/destination port and field set can be defined for TCP/UDP protocol. Also, fix the configuration order as per the eos running-config.

Reviewer 2: ```suggestion
As of AVD 5.0.0, the configuration of `flags` in `UDP` protocol under traffic-policies is not generated and the configuration for `source/destination port` and `field set` can be generated together for `TCP/UDP` protocol. Also, the configuration order has been fixed as per the EOS running-config.
```
